### ISSUE TITLE
Unable to determine environment

### ISSUE THREAD

**Issue by lencioni** [steelbrain/consistent-env#9]
I am developing an Atom plugin and trying to use this package to determine an env when calling external processes. 

My code looks like this:

``` js
const env = {};
consistentEnv.async()
  .then(computedEnv => { Object.assign(env, computedEnv); })
  .catch(e => { atom.notifications.addError(e); });
```

This seems to work maybe 50% of the time. The other ~50% I end up seeing the following error logged in the console:

```
[consistent-env] Unable to determine environment Error: Unable to determine environment
    at /path/to/node_modules/consistent-env/lib/helpers.js:71:11
```

If I change helpers.js:71 to throw the error it is catching

``` js
}).catch(function (error) {
  throw error;
  //throw new Error('Unable to determine environment');
});
```

then I get the following, more helpful message:

```
[consistent-env] Unable to determine environment Error: Process execution timed out
    at /path/to/node_modules/consistent-env/lib/helpers.js:58:14
```

Increasing the timeout from 2000 to 4000 seems to make it much better on my machine.

**Comment by steelbrain** [comment]
@lencioni from my experience with debugging this for users of linters so far, you may have a deadlock or something in your shell config. It can be listening for input on stdin

Can you please try with an empty shell config to see if you still get the error?

**Comment by steelbrain** [comment]
Also does this error happen consistently or occasionally?

If you are using zsh and using zsh compliments like `oh-my-zsh`, You could edit your `.zshrc` and make it only require the `oh-my-zsh` init files if `STDOUT` is a tty. There's a nice [`unix stackexchange answer`](http://unix.stackexchange.com/a/26782/133507) that covers this

**Comment by lencioni** [comment]
> Also does this error happen consistently or occasionally?

It seems to happen on one laptop fairly consistently, like maybe 40% of the time when I reload Atom. On my other laptop, which has the same configuration, it doesn't seem to happen.

I am using prezto, which might be causing some of the slowdown. I tried out the trick in that stackexchange answer (thanks!) and it seems to shave off about 300ms from the `consistentEnv.async()` operation on the laptop that wasn't too slow (1300ms vs 1600ms). On the laptop that was sometimes too slow, the results are more dramatic, bringing it down to a pretty stable ~850ms where before the change it was bouncing around from 1300ms to more than 2000ms.

Here's the exact configuration change I made if anyone is curious: https://github.com/lencioni/dotfiles/commit/3e2c0a884883d01aa900980a3a1493e6efcde619

I'm going to say case closed on this one. Thanks for the help!
